cmd/aws-s3-proxy/pkg/serve.go

```diff
@@ -1,67 +1,139 @@
-This application is a tool to generate the needed files
-to quickly create a Cobra application.`,
+	Short: "serve the s3 proxy",
 	Run: func(cmd *cobra.Command, args []string) {
 		serve()
 	},
 }
 
 func init() {
+	var (
+		idleConnTimeout        int
+		guessBucketTimeout     int
+		directoryListingFormat bool
+	)
+
+	c := config.Config
+	// Basic configs
+	cobra.OnInitialize(initConfig)
+	serveCmd.PersistentFlags().BoolVar(&c.AccessLog, "access-log", false, "toggle access log")
+	serveCmd.PersistentFlags().BoolVar(&c.AllPagesInDir, "get-all-pages-in-dir", false, "toggle getting all pages in directories")
+	serveCmd.PersistentFlags().BoolVar(&c.ContentEncoding, "content-access", true, "toggle content encoding")
+	serveCmd.PersistentFlags().BoolVar(&c.DirectoryListing, "directory-listing", false, "toggle directory listing")
+	serveCmd.PersistentFlags().BoolVar(&c.DisableCompression, "disable-compression", true, "toggle compression")
+	serveCmd.PersistentFlags().BoolVar(&c.DisableUpsteamSSL, "disable-upstream-ssl", false, "toggle tls for the aws-sdk")
+	serveCmd.PersistentFlags().StringVar(&c.HTTPCacheControl, "http-cache-control", "", "overrides S3's HTTP `Cache-Control` header")
+	serveCmd.PersistentFlags().StringVar(&c.HTTPExpires, "http-expires", "", "overrides S3's HTTP `Expires` header")
+	serveCmd.PersistentFlags().StringVar(&c.BasicAuthUser, "basic-auth-user", "", "username for basic auth")
+	serveCmd.PersistentFlags().StringVar(&c.SslCert, "ssl-cert-path", "", "path to ssl cert")
+	serveCmd.PersistentFlags().StringVar(&c.SslKey, "ssl-key-path", "", "path to ssl key")
+	serveCmd.PersistentFlags().StringVar(&c.StripPath, "strip-path", "", "strip path prefix")
+	serveCmd.PersistentFlags().StringVar(&c.CorsAllowOrigin, "cors-allow-origin", "", "CORS: a URI that may access the resource")
+	serveCmd.PersistentFlags().StringVar(&c.CorsAllowMethods, "cors-allow-methods", "", "CORS: comma-delimited list of the allowed - https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html")
+	serveCmd.PersistentFlags().StringVar(&c.CorsAllowHeaders, "cors-allow-headers", "", "CORS:Comma-delimited list of the supported request headers")
+	serveCmd.PersistentFlags().BoolVar(&c.InsecureTLS, "insecure-tls", false, "toggle insecure tls")
+	serveCmd.PersistentFlags().StringVar(&c.HealthCheckPath, "healthcheck-path", "", "path for healthcheck")
+	serveCmd.PersistentFlags().Int64Var(&c.CorsMaxAge, "cors-max-age", 600, "CORS: max age in seconds")
+	serveCmd.PersistentFlags().IntVar(&c.MaxIdleConns, "max-idle-connections", 150, "max idle connections")
+	serveCmd.PersistentFlags().StringVar(&c.Host, "listen-address", "::1", "host address to listen on")
+	serveCmd.PersistentFlags().StringVar(&c.IndexDocument, "index-document", "index.html", "the index document for static website")
+	serveCmd.PersistentFlags().StringVar(&c.Port, "list-port", "21080", "port to listen on")
+	serveCmd.PersistentFlags().StringVar(&c.S3Bucket, "upstream-bucket", "", "upstream s3 bucket")
```

Comment:
Does this need to be marked as required (`MarkPersistentFlagRequired`)? Seems like the old validation with fatal without this?